gatherers/manifest.js

```diff
@@ -54,34 +32,62 @@ class Manifest extends Gather {
     };
   }
 
+  /**
+   * Creates a listener for manifests in network requests, a load timeout, and
+   * a promise they resolve when complete.
+   * @private
+   */
+  static _resolveOnManifest() {
```

Comment:
Actually wondering if this code really belongs in the driver, where we can call getManifest, and we can resolve the promise in there? This just feels like a lot of boilerplate for something that we know we will need both drivers to do for us.